### DISCUSSION TITLE
Hardware wiring for matrix

### DISCUSSION THREAD

**Discussion by LoveMyKeys** [PlummersSoftwareLLC/NightDriverStrip#593] [Q&A] [answered]
Hi,

Does someone have a wiring diagram on how to connect M5Stick plus, 3 16*16  matrixes and a powersupply? The matrixes can be interconnected (and I have done so) but they also have seperate 5V and GND wire. Must these be connected to the powersupply? And the Stick, should the grove 5V and GND also be connected to the powersupply?

Thanks in advance.

**Comment by rbergen** [comment, marked as answer]
I don't have a diagram, but I can tell you the pinout I used - and share a picture of it.

First off I don't use the grove connector, but the female pin header at the other side of the M5StickC Plus. The reason is that I've also hooked up an IR receiver module to it, which requires 3.3V power, which is only available on the pin header. 

With that said, the pinout I use is:

- GND: connected to power supply ground and IR receiver ground pin (white)
- 5V out: disconnected
- G26: connected to LED matrix data line (green)
- G25/G36: connected to IR receiver data pin (blue)
- G0: disconnected
- BAT: disconnected
- 3V3: connected to IR receiver +3.3V pin (brown)
- 5V in: connected to the power supply 5V (red)

The colors refer to the wires in the following picture:

![PIN header](https://github.com/PlummersSoftwareLLC/NightDriverStrip/assets/10425218/d2a2ed36-0033-4054-8583-6393ead23f1c)

(The quality of the whole creation is why my day job is related to writing software, not creating hardware. :) )

Beyond that, I have indeed connected the power supply 5V and ground directly to the panels also.

I hope this helps.

**Reply by LoveMyKeys** [reply]
Thank you very much, now ordering the required connectors :)

**Comment by robertlipe** [comment]
Cool!

Rutger, what's the connector with only one wire connected and how does that
work?

If you ever plan to power 12/24VDC strips from this fixture, be sure that
the voltage from your barrel connector goes to your strip and only your
strip but the ground is shared with both that LED circuit and the
LED0/controller circuit. Then you can power things like
https://www.amazon.comdp/B0C458LYMB and not worry about dumping 24V into
your poor little 3.3v (there's an LDO on-board)  ESP32.


People working on projects like this should probably own a collection of
connectors like these
<https://www.amazon.com/KWMSTPLT-Connector-WS2812B-Soldering-Controller/dp/B0CH4KSQ6T>.
(There are other brands and vendors. It's just a part number I had handy.)
These make fixtures for hooking up strips and adding connectors to bare
strips for compatibility easy.

Anyone working on electronics should stock the 2.54mm headers in male and
female, single and dual row. These are just the rosetta stone of connecting
boards to peripherals, breadboards, proto boards, and removable headers.
Amazon has assortments you can get quickly, but if you're stocking a lab
and can wait, an order with Aliexpress is worth the wait. The crimp-on
connectors for wires to go in the blank shells are also worthwhile.

I should probably show-and-tell my hot-glue crafting nightmare project
here...it's not perfect, but it allows me to swap controller boards and
strips of varying voltages easily.

RJL

**Reply by LoveMyKeys** [reply]
I got the cable you mention along with the 3 matrixes so I have 3 of them now. Also ordered this from Amazon NL:

![IMG_1088](https://github.com/PlummersSoftwareLLC/NightDriverStrip/assets/154417166/e9221fe0-437e-4765-baee-69874b3e0e12)

**Reply by rbergen** [reply]
@robertlipe The green wire is LED (matrix) data. It suffices because I power the 16x16 matrices off the same power supply that I connect to the M5StickC Plus, which means both +5V and ground are shared across M5Stick and LEDs. I just chose to power the matrices directly via the separate power wires connected to them (to which I hooked up female barrel connectors for convenience), instead of the 3-prong connector that the data wire is connected to.

**Comment by robertlipe** [comment]
LoveMyKeys: Good call.  I have a very similar kit. Maybe even the same
crimpers. You won't have to stall a project waiting for one or hack up a
kludge because you don't have them for many years. Those have been JTAG,
Serial, LCD, SPI, I2C, and about every other imaginable connector for me
since I've started using them.

Rutger: Ah, now I get it.  Thanx for typing slower. I have just enough 12
and 24V strips that my designs keep them more strongly separated. SO FAR, I
haven't hit both buttons on my bench power supply (one to power up the
lights so I can dial in voltage and max current, one to actually unleash
the electrons) with it dialed at the wrong voltage or grabbed the wrong
(always poorly marked) brick from the pile and over-filled the electron
limits of a strip, but I know that day is coming. I considered committing
to three different plug types for 5, 12, and 24V but that meant instead of
it usually being right for some cases, I'd have to replace the ends on
everything and I just lacked that commitment. So I have a lot of 5525
connectors ... and a few 5521's just to be annoyingly incompatible. :-/ Now
I understand the way you did it though.
